### PR TITLE
Correct Security Definition link

### PR DESCRIPTION
**Pull request by jlurien**
#### What type of PR is this?

* correction

#### What this PR does / why we need it:

Fix incorrect reference

### CODE REVIEW

File: documentation/API-design-guidelines.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ This document captures guidelines for the API design in CAMARA project. These gu
       - [11.5.1 Usage of discriminator](#1151-usage-of-discriminator)
         - [Inheritance](#inheritance)
         - [Polymorphism](#polymorphism)
-    - [11.6 OAuth Definition](#116-oauth-definition)
+    - [11.6 Security Definition](116-security-definition)
```
Comment (jlurien) [inline documentation/API-design-guidelines.md:51]:
```suggestion
    - [11.6 Security Definition](#116-security-definition)
```

**Review by PedroDiez — APPROVED** [review]
LGTM

**Review by rartych — APPROVED** [review]
LGTM

**Review by patrice-conil — APPROVED** [review]
/LGTM

**Pull request merged**